src/mbed_tools/cli/build.py

```diff
@@ -82,9 +82,18 @@ def build(
        baudrate: Change the serial baud rate (ignored unless --sterm is also given).
     """
     if mbed_os_path is None:
-        program = MbedProgram.from_existing(pathlib.Path(program_path))
+        program = MbedProgram.from_existing(
+            pathlib.Path(program_path), target_name=mbed_target, profile=profile, toolchain=toolchain
+        )
     else:
-        program = MbedProgram.from_existing(pathlib.Path(program_path), pathlib.Path(mbed_os_path))
+        program = MbedProgram.from_existing(
+            pathlib.Path(program_path),
+            pathlib.Path(mbed_os_path),
+            target_name=mbed_target,
+            profile=profile,
+            toolchain=toolchain,
+        )
+
     mbed_config_file = program.files.cmake_config_file
     build_tree = program.files.cmake_build_dir
```

Comment:
Instead of passing target, toolchain and the kitchen sink to `MbedProgram` could we just construct the full `build_dir` path here? I think it would simplify this code a lot.